### PR TITLE
disallow case list form when module is configured to only show forms

### PR DESCRIPTION
**Pull request by snopoke**
@TylerSheffels

### CODE REVIEW

**Comment by snopoke** [comment]
Also updated the wiki

**Comment by snopoke** [comment]
hold off on merge, some more validation text to change

**Comment by snopoke** [comment]
GTG once tests pass

**Comment by TylerSheffels** [comment]
@snopoke Has conflicts now

File: corehq/apps/app_manager/suite_xml.py [hunk 1]
```diff
@@ -1277,57 +1277,9 @@ def build_detail(self, module, detail_type, detail, detail_column_infos,
...
-                                value=session_var(source_dm['datum'].id))
-                            )
-                    else:
-                        s_datum = target_meta['datum']
-                        frame.add_datum(StackDatum(id=s_datum.id, value=s_datum.function))
-
-                frame.add_datum(StackDatum(id=RETURN_TO, value=XPath.string(id_strings.menu_id(module))))
-                d.action.stack.add_frame(frame)
+            if module.case_list_form.form_id and detail_type.endswith('short')\
+                    and not module.put_in_root:
```
Comment (snopoke) [inline corehq/apps/app_manager/suite_xml.py]: Added in the `and not module.put_in_root`. The rest is just extracting it to a function.

**Comment by snopoke** [comment]
@TylerSheffels updated

**Pull request merged**